### ISSUE TITLE
Using data from API using request-promise throws an error

### ISSUE THREAD

**Issue by scottgruber** [frctl/fractal#234]
I'm trying to use dynamic data pulled from an API and ran into an error. This is my first time using a Promise and I've been doing some reading online but I'm very much a beginner. Hope it isn't something too obvious. Its probably unrelated to the [issue](https://github.com/frctl/fractal/issues/233) Matt posted yesterday but was surprised to see it raised when I was also running some tests on the same feature. I checked the issue board and searched in slack, but haven't run across something similar. Are there any other examples which uses data from an API? 

I've npm install request (2.81.0) and request-promise (4.2.0) and am running fractal 1.1.3 and mandelbrot 1.1.0

Here is the error after I run `gulp fractal:start`. 

```
Unhandled rejection TypeError: membersApiData[Symbol.iterator] is not a function
    at /Users/scottgruber/git/scottgruber/fractal/src/components/03-patterns/cards-hardboiled/cards-hardboiled.config.js:15:25
    at tryCatcher (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)

Unhandled rejection TypeError: membersApiData[Symbol.iterator] is not a function
    at /Users/scottgruber/git/scottgruber/fractal/src/components/03-patterns/cards-hardboiled/cards-hardboiled.config.js:15:25
    at tryCatcher (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/Users/scottgruber/git/scottgruber/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
```
Below are two variations of js code which throws the same error. One is the sample code with only a change in the uri. The second is an example adapted to the data model of the component.  I posted the hbs and js files in a gist [here](https://gist.github.com/scottgruber/8f5970c8bf7024e58af9ad95be1334e1) in case it helps. 

**Example 1**

```
 // member-list.config.js
 'use strict';

 const request = require('request-promise'); // require the request-promise module

 // make the request to the API, returns a Promise
 const response = request({
     uri: 'https://scottgruber.me/api/hardboiled',
     json: true
 });

 // do some post-processing on the response to wrangle it into the correct format
 response.then(function (membersApiData) {
     const memberData = [];
     for (let member of membersApiData) {
         memberData.push({
             name: `${member.firstName} ${member.lastName}`,
             email: member.emailAddress
         });
     }
     return memberData;
 });

 module.exports = {
     context: {
         members: response // use the response as context data for our template.
     }
 };

```

**Example 2**

```
// cards-hardboiled.config.js
'use strict';

const request = require('request-promise'); // require the request-promise module

// make the request to the API, returns a Promise
const response = request({
    uri: 'https://scottgruber.me/api/hardboiled',
    json: true
});

// do some post-processing on the response to wrangle it into the correct format
response.then(function(hardboiledApiData) {

    const bookData = [];
    for (let book of hardboiledApiData) {
        bookData.push({
            title: book.heading,
            subhead: book.subhead,
            author: book.author,
            image: book.image,
            text: book.body
        });
    }
    return bookData;
});

module.exports = {
    context: {
        book: response // use the response as context data for our template.
    }
};

```

**Comment by cfinucane** [comment]
I'm not super familiar with ES6, but I think `for...of` iteration only works by default with arrays, and your API returns an object.  (See https://stackoverflow.com/questions/29885220/using-objects-in-for-of-loops)

**Comment by dkhuntrods** [comment]
@scottgruber I'd agree with @cfinucane, that looks like an error on your iterator. 
I'd also suggest removing your `node_modules` directory and updating Fractal to 1.1.4, just to prevent any other issues, as there's an important fix in there (see #232) .
I'm going to close this as it doesn't look related to Fractal specifically, but if you've got any more Fractal-specific issues do please open a new one :)

**Comment by nfourtythree** [comment]
I am stilling getting an issue with using data from an API and also the issue mentioned in #232

I can get the promise to log the data from API (it seems to log the data twice). But as soon as I add the promise into the `context` it seems to fall over.

Have tried re-installing the node modules and completely killing the folder etc. Have also tried manually changing the `vorpal` version. Still no luck

Any thoughts? 🤔
